### ISSUE TITLE
Permissions?

### ISSUE THREAD

**Issue by dodekeract** [tomav/docker-mailserver#604]
## Problem

I tried to secure my emails against other Linux users, so I did a `chown -R dodekeract:dodekeract mailserver/` and a `chmod -R a-rwx mailserver/`.

`mailserver/` contains `config`, `data` and `state`. Now it doesn't work anymore. 😞

Any idea which permissions are required and how I can manage to only let my user view the data?

## Logs

<details>
<summary>docker logs mail (censored email address)</summary>

```
May 13 15:13:56 mail dovecot: imap(mail@email.address.censored): Error: Invalid user settings. Refer to server log for more information.
May 13 15:13:58 mail dovecot: imap-login: Login: user=<mail@email.address.censored>, method=PLAIN, rip=79.247.178.231, lip=172.18.0.2, mpid=10783, TLS, session=<aYhyRWlPOABP97Ln>
May 13 15:13:58 mail dovecot: imap(mail@email.address.censored): Error: chdir(/var/mail/email.address.censored/mail/) failed: Permission denied (euid=5000(docker) egid=5000(docker) missing +x perm: /var/mail, we're not in group 1000, dir owned by 1000:1000 mode=0750)
May 13 15:13:58 mail dovecot: imap(mail@email.address.censored): Error: chdir(/var/mail/email.address.censored/mail) failed: Permission denied
May 13 15:13:58 mail dovecot: imap(mail@email.address.censored): Error: user mail@email.address.censored: Initialization failed: Namespace '': stat(/var/mail/email.address.censored/mail) failed: Permission denied (euid=5000(docker) egid=5000(docker) missing +x perm: /var/mail, we're not in group 1000, dir owned by 1000:1000 mode=0750)
May 13 15:13:58 mail dovecot: imap(mail@email.address.censored): Error: Invalid user settings. Refer to server log for more information.
May 13 15:14:00 mail dovecot: imap-login: Login: user=<mail@email.address.censored>, method=PLAIN, rip=79.247.178.231, lip=172.18.0.2, mpid=10785, TLS, session=<XxCTRWlPPgBP97Ln>
May 13 15:14:00 mail dovecot: imap(mail@email.address.censored): Error: chdir(/var/mail/email.address.censored/mail/) failed: Permission denied (euid=5000(docker) egid=5000(docker) missing +x perm: /var/mail, we're not in group 1000, dir owned by 1000:1000 mode=0750)
May 13 15:14:00 mail dovecot: imap(mail@email.address.censored): Error: chdir(/var/mail/email.address.censored/mail) failed: Permission denied
May 13 15:14:00 mail dovecot: imap(mail@email.address.censored): Error: user mail@email.address.censored: Initialization failed: Namespace '': stat(/var/mail/email.address.censored/mail) failed: Permission denied (euid=5000(docker) egid=5000(docker) missing +x perm: /var/mail, we're not in group 1000, dir owned by 1000:1000 mode=0750)
May 13 15:14:00 mail dovecot: imap(mail@email.address.censored): Error: Invalid user settings. Refer to server log for more information.
May 13 15:14:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:14:23 mail postfix/qmgr[10786]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:14:24 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10786 exit status 1
May 13 15:14:24 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:15:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:15:24 mail postfix/qmgr[10787]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:15:25 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10787 exit status 1
May 13 15:15:25 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:15:34 mail postfix/smtpd[10788]: connect from o1.sgmail.github.com[192.254.114.176]
May 13 15:15:34 mail postfix/smtpd[10788]: warning: connect to private/anvil: Permission denied
May 13 15:15:34 mail postfix/smtpd[10788]: warning: problem talking to server private/anvil: Permission denied
May 13 15:15:35 mail postfix/smtpd[10788]: warning: connect to private/anvil: Permission denied
May 13 15:15:35 mail postfix/smtpd[10788]: warning: problem talking to server private/anvil: Permission denied
May 13 15:15:35 mail postfix/smtpd[10788]: warning: SASL: Connect to /var/spool/postfix/private/auth failed: Permission denied
May 13 15:15:35 mail postfix/smtpd[10788]: fatal: no SASL authentication mechanisms
May 13 15:15:36 mail postfix/master[988]: warning: process /usr/lib/postfix/smtpd pid 10788 exit status 1
May 13 15:15:36 mail postfix/master[988]: warning: /usr/lib/postfix/smtpd: bad command startup -- throttling
May 13 15:16:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:16:25 mail postfix/qmgr[10789]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:16:26 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10789 exit status 1
May 13 15:16:26 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:17:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:17:26 mail postfix/qmgr[10793]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:17:27 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10793 exit status 1
May 13 15:17:27 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:17:46 mail postfix/smtpd[10809]: warning: hostname no-reverse-dns-configured.com does not resolve to address 80.82.65.127
May 13 15:17:46 mail postfix/smtpd[10809]: connect from unknown[80.82.65.127]
May 13 15:17:46 mail postfix/smtpd[10809]: warning: connect to private/anvil: Permission denied
May 13 15:17:46 mail postfix/smtpd[10809]: warning: problem talking to server private/anvil: Permission denied
May 13 15:17:47 mail postfix/smtpd[10809]: warning: connect to private/anvil: Permission denied
May 13 15:17:47 mail postfix/smtpd[10809]: warning: problem talking to server private/anvil: Permission denied
May 13 15:17:47 mail postfix/smtpd[10809]: warning: SASL: Connect to /var/spool/postfix/private/auth failed: Permission denied
May 13 15:17:47 mail postfix/smtpd[10809]: fatal: no SASL authentication mechanisms
May 13 15:17:48 mail postfix/master[988]: warning: process /usr/lib/postfix/smtpd pid 10809 exit status 1
May 13 15:17:48 mail postfix/master[988]: warning: /usr/lib/postfix/smtpd: bad command startup -- throttling
May 13 15:18:05 mail postfix/sendmail[10821]: fatal: chdir /var/spool/postfix: Permission denied
May 13 15:18:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:18:27 mail postfix/qmgr[10823]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:18:28 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10823 exit status 1
May 13 15:18:28 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:18:48 mail postfix/smtpd[10826]: connect from o7.sgmail.github.com[167.89.101.198]
May 13 15:18:48 mail postfix/smtpd[10826]: warning: connect to private/anvil: Permission denied
May 13 15:18:48 mail postfix/smtpd[10826]: warning: problem talking to server private/anvil: Permission denied
May 13 15:18:49 mail postfix/smtpd[10826]: warning: connect to private/anvil: Permission denied
May 13 15:18:49 mail postfix/smtpd[10826]: warning: problem talking to server private/anvil: Permission denied
May 13 15:18:49 mail postfix/smtpd[10826]: warning: SASL: Connect to /var/spool/postfix/private/auth failed: Permission denied
May 13 15:18:49 mail postfix/smtpd[10826]: fatal: no SASL authentication mechanisms
May 13 15:18:50 mail postfix/master[988]: warning: process /usr/lib/postfix/smtpd pid 10826 exit status 1
May 13 15:18:50 mail postfix/master[988]: warning: /usr/lib/postfix/smtpd: bad command startup -- throttling
May 13 15:19:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:19:28 mail postfix/qmgr[10830]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:19:29 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10830 exit status 1
May 13 15:19:29 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:20:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:20:29 mail postfix/qmgr[10831]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:20:30 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10831 exit status 1
May 13 15:20:30 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:20:37 mail postfix/smtpd[10832]: connect from github-smtp2-ext2.iad.github.net[192.30.252.193]
May 13 15:20:37 mail postfix/smtpd[10832]: warning: connect to private/anvil: Permission denied
May 13 15:20:37 mail postfix/smtpd[10832]: warning: problem talking to server private/anvil: Permission denied
May 13 15:20:38 mail postfix/smtpd[10832]: warning: connect to private/anvil: Permission denied
May 13 15:20:38 mail postfix/smtpd[10832]: warning: problem talking to server private/anvil: Permission denied
May 13 15:20:38 mail postfix/smtpd[10832]: warning: SASL: Connect to /var/spool/postfix/private/auth failed: Permission denied
May 13 15:20:38 mail postfix/smtpd[10832]: fatal: no SASL authentication mechanisms
May 13 15:20:39 mail postfix/master[988]: warning: process /usr/lib/postfix/smtpd pid 10832 exit status 1
May 13 15:20:39 mail postfix/master[988]: warning: /usr/lib/postfix/smtpd: bad command startup -- throttling
May 13 15:21:21 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:21:30 mail postfix/qmgr[10834]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:21:31 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10834 exit status 1
May 13 15:21:31 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:21:39 mail postfix/smtpd[10835]: warning: hostname dynamic-ip-181550176241.cable.net.co does not resolve to address 181.55.176.241: Name or service not known
May 13 15:21:39 mail postfix/smtpd[10835]: connect from unknown[181.55.176.241]
May 13 15:21:39 mail postfix/smtpd[10835]: warning: connect to private/anvil: Permission denied
May 13 15:21:39 mail postfix/smtpd[10835]: warning: problem talking to server private/anvil: Permission denied
May 13 15:21:40 mail postfix/smtpd[10835]: warning: connect to private/anvil: Permission denied
May 13 15:21:40 mail postfix/smtpd[10835]: warning: problem talking to server private/anvil: Permission denied
May 13 15:21:40 mail postfix/smtpd[10835]: warning: SASL: Connect to /var/spool/postfix/private/auth failed: Permission denied
May 13 15:21:40 mail postfix/smtpd[10835]: fatal: no SASL authentication mechanisms
May 13 15:21:41 mail postfix/master[988]: warning: process /usr/lib/postfix/smtpd pid 10835 exit status 1
May 13 15:21:41 mail postfix/master[988]: warning: /usr/lib/postfix/smtpd: bad command startup -- throttling
May 13 15:22:22 mail postfix/master[988]: warning: master_wakeup_timer_event: service pickup(public/pickup): Permission denied
May 13 15:22:31 mail postfix/qmgr[10836]: fatal: scan_dir_push: open directory active: Permission denied
May 13 15:22:32 mail postfix/master[988]: warning: process /usr/lib/postfix/qmgr pid 10836 exit status 1
May 13 15:22:32 mail postfix/master[988]: warning: /usr/lib/postfix/qmgr: bad command startup -- throttling
May 13 15:22:41 mail postfix/smtpd[10837]: connect from unknown[195.22.126.189]
May 13 15:22:41 mail postfix/smtpd[10837]: warning: connect to private/anvil: Permission denied
May 13 15:22:41 mail postfix/smtpd[10837]: warning: problem talking to server private/anvil: Permission denied
May 13 15:22:42 mail postfix/smtpd[10837]: warning: connect to private/anvil: Permission denied
May 13 15:22:42 mail postfix/smtpd[10837]: warning: problem talking to server private/anvil: Permission denied
May 13 15:22:42 mail postfix/smtpd[10837]: warning: SASL: Connect to /var/spool/postfix/private/auth failed: Permission denied
May 13 15:22:42 mail postfix/smtpd[10837]: fatal: no SASL authentication mechanisms
```
</details>

## Why Reopen?

~~Closed, apparently the permissions fix themselves automatically as soon as you restart the server. Didn't expect that. Maybe it helps people googling for this though. :)~~ [relevant line](https://github.com/tomav/docker-mailserver/blob/c0530491c39873369d2fdf1540b180dd99043ad5/target/start-mailserver.sh#L1033)

Re-opening as sending and receiving mails doesn't work anymore. `docker logs -f mail` doesn't show any errors though.

## What I found out so far

1. After restarting, the `mail`/`data` folder gets fixed. That's a good start and allows me to read my mails.

2. `state` can be deleted. I didn't configure anything there, so that folder is fixable too.

So basically this whole issue can be resolved if anyone could share a `ls -lAh mailserver/config`. I think I only need those permissions.

3. I also managed to get it to work again. I can at least receive/send mails again. However, I had to add `chmod -R o+rwx` to my `config` folder, which is a security nightmare. Can someone clarify which permissions should be set where?

**Comment by dodekeract** [comment]
Closed, apparently the permissions fix themselves automatically as soon as you restart the server.

Didn't expect that. Maybe it helps people googling for this though. :)

**Comment by techgourmet** [comment]
Something is seriously wrong with this last build.

- No delivery of email, postfix errors in syslog (permission denied) rendering it unusable.

**Comment by tomav** [comment]
Duplicates #545
Move this discussion there.
